src/models/document/document-content.ts

```diff
@@ -730,27 +680,32 @@ export const DocumentContentModel = types
         }
       },
       addTile(tool: DocumentTool, options?: IDocumentContentAddTileOptions) {
-        const { title, addSidecarNotes, url, insertRowInfo } = options || {};
+        const { title, addSidecarNotes, url, text, insertRowInfo } = options || {};
         // for historical reasons, this function initially places new rows at
         // the end of the content and then moves them to the desired location.
         const addTileOptions = { rowIndex: self.rowCount };
-        let tileInfo;
-        switch (tool) {
-          case "text":
-            tileInfo = self.addTextTile(addTileOptions);
-            break;
-          case "table":
-            tileInfo = self.addTableTile({ title, ...addTileOptions });
-            break;
-          case "geometry":
-            tileInfo = self.addGeometryTile({ title, addSidecarNotes, ...addTileOptions });
-            break;
-          case "image":
-            tileInfo = self.addImageTile({ url, ...addTileOptions });
-            break;
-          case "drawing":
-            tileInfo = self.addDrawingTile(addTileOptions);
-            break;
+        const contentInfo = getToolContentInfoByTool(tool);
+        const documents = getParentWithTypeName(self, "Documents") as DocumentsModelType;
+        const unit = documents?.unit;
```

Comment:
It'd be good to add an API for this instead of just passing the unit model.

It is possible to disable features at problem or unit level, that goes through the stores.
For example linking tables to graphs is disabled in one problem.

Reply:
Maybe could put the stamps in store. So the content model can access it without it being passed in.  This works because there is only one unit active.

Reply:
I'm going to make a new PR with a document including these ideas so we don't lose track of them.